src/MobileRedirection.php

```diff
@@ -165,21 +177,59 @@ public function redirect() {
 			}
 
 			// Add a link to the footer to allow for navigation to the AMP version.
-			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+			$this->add_mobile_switcher_footer_hooks( false );
```

Comment:
See note about how I don't think the `$legacy` arg is necessary.

```suggestion
			$this->add_mobile_switcher_footer_hooks();
```